datafusion/core/src/datasource/listing/table.rs

```
@@ -1716,6 +1719,9 @@ mod tests {
                     )
                     .await?;
             }
+            FileType::Extension(_) => {
+                panic!("Extension file type not implemented in write path.")
```

Can we instead return an NotImplement exception?